solidity/contracts/Gravity.sol

```diff
@@ -180,6 +191,31 @@ contract Gravity is ReentrancyGuard {
 		}
 	}
 
+	// Utility function to check for duplicate validators
+	// The checks are performed in memory and discarded
+	function checkDuplicateValidators(address[] memory _validators) private {
+		ValidatorPosition nextPosition;
+
+		if (
+			currentPosition == ValidatorPosition.UNINITIALIZED ||
+			nextPosition == ValidatorPosition.RIGHT
+		) {
+			nextPosition = ValidatorPosition.LEFT;
+		} else {
+			nextPosition = ValidatorPosition.RIGHT;
+		}
+
+		for (uint256 i = 0; i < _validators.length; i++) {
```

Comment:
What if the length of "_validators" is less than validatorSet? As a result, the "validatorSet" will be with somve validators wich were in the previous set, right?